GitCommands/RevisionReader.cs

```diff
@@ -172,44 +172,46 @@ public sealed class RevisionReader : IDisposable
             string revisionFilter,
             string pathFilter)
         {
-            return new GitArgumentBuilder("log")
+            var builder = new GitArgumentBuilder("log");
+            builder.Add("-z");
+            builder.Add($"--pretty=format:\"{FullFormat}\"");
+            if (!string.IsNullOrWhiteSpace(branchFilter) && isBranchFilterSimple(branchFilter))
             {
-                "-z",
-                branchFilter,
-                $"--pretty=format:\"{FullFormat}\"",
+                builder.Add(branchFilter);
+            }
```

Comment:
The original syntax was terse, and thus I'm finding it a bit more readable.

What's wrong with the following?
```cs
            return new GitArgumentBuilder("log")
            {
                "-z",
                branchFilter,
                { !string.IsNullOrWhiteSpace(branchFilter) && isBranchFilterSimple(branchFilter), branchFilter }
            }
```

Reply:
The sequence `"-z", branchFilter` is only valid if the branch filter is simple

```bash
# valid
git log -z full_branch_name
git log --branches=*_name

#invalid
git log -z *-name
git log -z "*-name"
```

Reply:
thus I need the `if` which breaks initializer-like syntax:
```
if (is_simple_branch_filter)
  builder.Add(branchFilter);
```

Reply:
Somehow I missed the possibility to
```
{boolean_condition, the_flag}
```
I was under impression, that the only possibility of conditional flag was ternary semantics
```
{boolean_condition, flag_if_true, flat_if_false}
```

Reply:
fixed